index.ts

```diff
@@ -14,53 +14,45 @@ const cachedAxios = axios.create({ adapter: cache.adapter });
 
 export default async (req: NowRequest, res: NowResponse) => {
   try {
-    const format =
-      req.url && req.url.substr(req.url.length - 3) === "xml" ? "xml" : "html";
+    let format = "html";
+    if (req.url && req.url.includes(".xml")) {
+      format = "xml";
+      res.setHeader("Content-Type", "application/atom+xml");
+    }
 
-    const object = sanitizeUrl(req.url);
-    if (!object || object.username.length == 0) {
+    const username = fetchUsername(req.url);
+    if (!username) {
       throw new Error(
-        "provide page @username in url, eg. https://calmbook.page/TurismoArgentina"
+        "provide page username in url, eg. https://calmbook.page/TurismoArgentina"
       );
     }
 
     const response = await cachedAxios.get(
-      object.itemid
-        ? `https://www.facebook.com/${object.username}/posts/${
-            object.itemid
-          }?_fb_noscript=1`
-        : `https://www.facebook.com/${object.username}/posts`
+      `https://www.facebook.com/${username}/posts`
     );
 
     const $ = cheerio.load(response.data);
-    const pageInfo = buildPage($);
-    const page: Page = { username: object.username, ...pageInfo };
+    const page: Page = buildPage($);
 
     const render = pug.compileFile(`${__dirname}/views/page.${format}.pug`);
-    const output = render({ page, moment });
-
-    if (format === "xml") {
-      res.setHeader("Content-Type", "application/atom+xml");
-    }
+    const output = render({ page, username, moment });
```

Comment:
Why do you pass `username` as a separate variable and not as a page property (which `username` is)?

Reply:
My reasoning for having it _global_ is:

* You find page by username, so I understand hierarchy more like this: username -> page -> posts. 
* I found it more complex to fetch page by username and then to explicitly push username into page. 
* I found it confusing that `buildPage` was building and returning incomplete page (without username).

Reply:
Saying that, we still can make `username` part of Page interface, but imho it'll be much cleaner if we do it by passing `username` to `buildPage` OR making `buildPage` to fetch it on its own.

Reply:
> My reasoning for having it _global_ is:
> 
> * You find page by username, so I understand hierarchy more like this: username -> page -> posts.
> * I found it more complex to fetch page by username and then to explicitly push username into page.
> * I found it confusing that `buildPage` was building and returning incomplete page (without username).

I fully agree with third point - using `Omit<>` makes you feel bad :)

Reply:
The solution to modify `buildPost` to fetch username from page code still seems weird.

The best solution would bo to rename `buildPost` to `buildPostInfo` (or `buildPostData`) and keep hierarchy like : username -> page_info -> posts.

Reply:
@sobstel ⬆️lets use this hierarchy at this point and move forward

Reply:
As a matter of fact, this is not even related to what this PR is primarily about 🙈 That's why I'm going to revert page-related change and move all the entertaining heat of this discussion somewhere else ;-)